File: src/main/java/destiny/astrology/classical/rules/debilities/EssentialRule.java
/**
 * @author smallufo
 * Created on 2007/12/30 at 上午 4:48:47
 */
package destiny.astrology.classical.rules.debilities;

import destiny.astrology.classical.EssentialDefaultImpl;
import destiny.astrology.classical.EssentialIF;

abstract class EssentialRule extends Rule {

  /** 具備計算 Ptolemy's Table of Essential Dignities and Debilities 的所有介面 */
  EssentialIF essentialImpl = new EssentialDefaultImpl();

  EssentialRule() {
    super();
  }

  public EssentialIF getEssentialImpl() {
    return essentialImpl;
  }

  public void setEssentialImpl(EssentialIF essentialImpl) {
    this.essentialImpl = essentialImpl;
  }


}

File: src/main/java/destiny/astrology/classical/rules/debilities/EssentialRule.kt
/**
 * @author smallufo
 * Created on 2007/12/30 at 上午 4:48:47
 */
package destiny.astrology.classical.rules.debilities

import destiny.astrology.classical.*

abstract class EssentialRule : DebilityRule() {

  val triplicityImpl : ITriplicity = TriplicityWilliamImpl()

  val rulerImpl : IRuler = RulerPtolemyImpl()
  val detrimentImpl : IDetriment = DetrimentPtolemyImpl()
  val exaltImpl : IExaltation = ExaltationPtolemyImpl()
  val fallImpl : IFall = FallPtolemyImpl()

  val termImpl: ITerm = TermPtolomyImpl()
  val faceImpl : IFace = FacePtolomyImpl()

}
